src/main/java/org/folio/rest/client/impl/UserModuleClientImpl.java

```diff
@@ -34,10 +37,73 @@ public UserModuleClientImpl(HttpClient httpClient) {
           case HttpStatus.SC_NOT_FOUND:
             return Optional.empty();
           default:
-            String logMessage =
-              String.format("Error looking up for user. Status: %d, body: %s", response.getCode(), response.getBody());
-            throw new OkapiModuleClientException(logMessage);
+            throw new OkapiModuleClientException(generateErrorLogMsg(response));
+        }
+      });
+  }
+
+  @Override
+  public Future<Optional<User>> lookupUserByUserName(String userName, OkapiConnectionParams connectionParams) {
+    String requestUrl = connectionParams.getOkapiUrl() + "/users?query=username==\"" + userName + "\"";
```

Comment:
The userName should be cql encoded, the query should be percent or urlencoded:
https://github.com/folio-org/raml-module-builder/blob/v32.2.2/util/src/main/java/org/folio/util/StringUtil.java#L65

```
String requestUrl = connectionParams.getOkapiUrl() + "/users?query="
  + StringUtil.urlEncode("username==" + StringUtil.cqlEncode(userName));
```